### ISSUE TITLE
Cannot install via pip on Python 3.10

### ISSUE THREAD

**Issue by robd003** [MagicStack/uvloop#417]
* **uvloop version**: 0.15.2
* **Python version**: 3.10b3
* **Platform**: macOS
* **Can you reproduce the bug with `PYTHONASYNCIODEBUG` in env?**: yes
* **Does uvloop behave differently from vanilla asyncio? How?**: Won't build

When trying to build uvloop against Python 3.10b3 the following error is emitted if you install via pip:

```
    ERROR: Command errored out with exit status 1:
     command: /Users/robd/code/allfactors-sanic/env/bin/python3.10 -u -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'/private/var/folders/_1/hjn676yd5kq4ym6ksh47ns8h0000gn/T/pip-install-bt4385ls/uvloop_5ce278aa65a2427aa3ecbb25f01a5a84/setup.py'"'"'; __file__='"'"'/private/var/folders/_1/hjn676yd5kq4ym6ksh47ns8h0000gn/T/pip-install-bt4385ls/uvloop_5ce278aa65a2427aa3ecbb25f01a5a84/setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record /private/var/folders/_1/hjn676yd5kq4ym6ksh47ns8h0000gn/T/pip-record-hgw3w18r/install-record.txt --single-version-externally-managed --compile --install-headers /Users/robd/code/allfactors-sanic/env/include/site/python3.10/uvloop
         cwd: /private/var/folders/_1/hjn676yd5kq4ym6ksh47ns8h0000gn/T/pip-install-bt4385ls/uvloop_5ce278aa65a2427aa3ecbb25f01a5a84/
    Complete output (103 lines):
    running install
    running build
    running build_py
    creating build/lib.macosx-11.0-x86_64-3.10
    creating build/lib.macosx-11.0-x86_64-3.10/uvloop
    copying uvloop/_version.py -> build/lib.macosx-11.0-x86_64-3.10/uvloop
    copying uvloop/_noop.py -> build/lib.macosx-11.0-x86_64-3.10/uvloop
    copying uvloop/_testbase.py -> build/lib.macosx-11.0-x86_64-3.10/uvloop
    copying uvloop/__init__.py -> build/lib.macosx-11.0-x86_64-3.10/uvloop
    running egg_info
    writing uvloop.egg-info/PKG-INFO
    writing dependency_links to uvloop.egg-info/dependency_links.txt
    writing requirements to uvloop.egg-info/requires.txt
    writing top-level names to uvloop.egg-info/top_level.txt
    reading manifest file 'uvloop.egg-info/SOURCES.txt'
    reading manifest template 'MANIFEST.in'
    warning: no previously-included files matching '*' found under directory 'vendor/libuv/.git'
    warning: no previously-included files matching '*' found under directory 'vendor/libuv/docs'
    warning: no previously-included files matching '*' found under directory 'vendor/libuv/img'
    adding license file 'LICENSE-APACHE'
    adding license file 'LICENSE-MIT'
    writing manifest file 'uvloop.egg-info/SOURCES.txt'
    copying uvloop/cbhandles.pxd -> build/lib.macosx-11.0-x86_64-3.10/uvloop
    copying uvloop/cbhandles.pyx -> build/lib.macosx-11.0-x86_64-3.10/uvloop
    copying uvloop/dns.pyx -> build/lib.macosx-11.0-x86_64-3.10/uvloop
    copying uvloop/errors.pyx -> build/lib.macosx-11.0-x86_64-3.10/uvloop
    copying uvloop/loop.c -> build/lib.macosx-11.0-x86_64-3.10/uvloop
    copying uvloop/loop.pxd -> build/lib.macosx-11.0-x86_64-3.10/uvloop
    copying uvloop/loop.pyi -> build/lib.macosx-11.0-x86_64-3.10/uvloop
    copying uvloop/loop.pyx -> build/lib.macosx-11.0-x86_64-3.10/uvloop
    copying uvloop/lru.pyx -> build/lib.macosx-11.0-x86_64-3.10/uvloop
    copying uvloop/pseudosock.pyx -> build/lib.macosx-11.0-x86_64-3.10/uvloop
    copying uvloop/py.typed -> build/lib.macosx-11.0-x86_64-3.10/uvloop
    copying uvloop/request.pxd -> build/lib.macosx-11.0-x86_64-3.10/uvloop
    copying uvloop/request.pyx -> build/lib.macosx-11.0-x86_64-3.10/uvloop
    copying uvloop/server.pxd -> build/lib.macosx-11.0-x86_64-3.10/uvloop
    copying uvloop/server.pyx -> build/lib.macosx-11.0-x86_64-3.10/uvloop
    copying uvloop/sslproto.pxd -> build/lib.macosx-11.0-x86_64-3.10/uvloop
    copying uvloop/sslproto.pyx -> build/lib.macosx-11.0-x86_64-3.10/uvloop
    creating build/lib.macosx-11.0-x86_64-3.10/uvloop/handles
    copying uvloop/handles/async_.pxd -> build/lib.macosx-11.0-x86_64-3.10/uvloop/handles
    copying uvloop/handles/async_.pyx -> build/lib.macosx-11.0-x86_64-3.10/uvloop/handles
    copying uvloop/handles/basetransport.pxd -> build/lib.macosx-11.0-x86_64-3.10/uvloop/handles
    copying uvloop/handles/basetransport.pyx -> build/lib.macosx-11.0-x86_64-3.10/uvloop/handles
    copying uvloop/handles/check.pxd -> build/lib.macosx-11.0-x86_64-3.10/uvloop/handles
    copying uvloop/handles/check.pyx -> build/lib.macosx-11.0-x86_64-3.10/uvloop/handles
    copying uvloop/handles/handle.pxd -> build/lib.macosx-11.0-x86_64-3.10/uvloop/handles
    copying uvloop/handles/handle.pyx -> build/lib.macosx-11.0-x86_64-3.10/uvloop/handles
    copying uvloop/handles/idle.pxd -> build/lib.macosx-11.0-x86_64-3.10/uvloop/handles
    copying uvloop/handles/idle.pyx -> build/lib.macosx-11.0-x86_64-3.10/uvloop/handles
    copying uvloop/handles/pipe.pxd -> build/lib.macosx-11.0-x86_64-3.10/uvloop/handles
    copying uvloop/handles/pipe.pyx -> build/lib.macosx-11.0-x86_64-3.10/uvloop/handles
    copying uvloop/handles/poll.pxd -> build/lib.macosx-11.0-x86_64-3.10/uvloop/handles
    copying uvloop/handles/poll.pyx -> build/lib.macosx-11.0-x86_64-3.10/uvloop/handles
    copying uvloop/handles/process.pxd -> build/lib.macosx-11.0-x86_64-3.10/uvloop/handles
    copying uvloop/handles/process.pyx -> build/lib.macosx-11.0-x86_64-3.10/uvloop/handles
    copying uvloop/handles/stream.pxd -> build/lib.macosx-11.0-x86_64-3.10/uvloop/handles
    copying uvloop/handles/stream.pyx -> build/lib.macosx-11.0-x86_64-3.10/uvloop/handles
    copying uvloop/handles/streamserver.pxd -> build/lib.macosx-11.0-x86_64-3.10/uvloop/handles
    copying uvloop/handles/streamserver.pyx -> build/lib.macosx-11.0-x86_64-3.10/uvloop/handles
    copying uvloop/handles/tcp.pxd -> build/lib.macosx-11.0-x86_64-3.10/uvloop/handles
    copying uvloop/handles/tcp.pyx -> build/lib.macosx-11.0-x86_64-3.10/uvloop/handles
    copying uvloop/handles/timer.pxd -> build/lib.macosx-11.0-x86_64-3.10/uvloop/handles
    copying uvloop/handles/timer.pyx -> build/lib.macosx-11.0-x86_64-3.10/uvloop/handles
    copying uvloop/handles/udp.pxd -> build/lib.macosx-11.0-x86_64-3.10/uvloop/handles
    copying uvloop/handles/udp.pyx -> build/lib.macosx-11.0-x86_64-3.10/uvloop/handles
    creating build/lib.macosx-11.0-x86_64-3.10/uvloop/includes
    copying uvloop/includes/__init__.py -> build/lib.macosx-11.0-x86_64-3.10/uvloop/includes
    copying uvloop/includes/compat.h -> build/lib.macosx-11.0-x86_64-3.10/uvloop/includes
    copying uvloop/includes/consts.pxi -> build/lib.macosx-11.0-x86_64-3.10/uvloop/includes
    copying uvloop/includes/debug.h -> build/lib.macosx-11.0-x86_64-3.10/uvloop/includes
    copying uvloop/includes/debug.pxd -> build/lib.macosx-11.0-x86_64-3.10/uvloop/includes
    copying uvloop/includes/flowcontrol.pxd -> build/lib.macosx-11.0-x86_64-3.10/uvloop/includes
    copying uvloop/includes/fork_handler.h -> build/lib.macosx-11.0-x86_64-3.10/uvloop/includes
    copying uvloop/includes/python.pxd -> build/lib.macosx-11.0-x86_64-3.10/uvloop/includes
    copying uvloop/includes/stdlib.pxi -> build/lib.macosx-11.0-x86_64-3.10/uvloop/includes
    copying uvloop/includes/system.pxd -> build/lib.macosx-11.0-x86_64-3.10/uvloop/includes
    copying uvloop/includes/uv.pxd -> build/lib.macosx-11.0-x86_64-3.10/uvloop/includes
    running build_ext
    building 'uvloop.loop' extension
    creating build/temp.macosx-11.0-x86_64-3.10
    creating build/temp.macosx-11.0-x86_64-3.10/uvloop
    clang -Wno-unused-result -Wsign-compare -Wunreachable-code -fno-common -dynamic -DNDEBUG -g -fwrapv -O3 -Wall -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk -I/usr/local/include -I/usr/local/opt/openssl@1.1/include -I/usr/local/opt/sqlite/include -I/usr/local/opt/tcl-tk/include -I/Users/robd/code/allfactors-sanic/env/include -I/usr/local/Cellar/python@3.10/3.10.0b3/Frameworks/Python.framework/Versions/3.10/include/python3.10 -I/private/var/folders/_1/hjn676yd5kq4ym6ksh47ns8h0000gn/T/pip-install-bt4385ls/uvloop_5ce278aa65a2427aa3ecbb25f01a5a84/vendor/libuv/include -c uvloop/loop.c -o build/temp.macosx-11.0-x86_64-3.10/uvloop/loop.o -O2
    uvloop/loop.c:169806:19: error: implicit declaration of function '_PyGen_Send' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
                ret = _PyGen_Send((PyGenObject*)yf, value == Py_None ? NULL : value);
                      ^
    uvloop/loop.c:169806:17: warning: incompatible integer to pointer conversion assigning to 'PyObject *' (aka 'struct _object *') from 'int' [-Wint-conversion]
                ret = _PyGen_Send((PyGenObject*)yf, value == Py_None ? NULL : value);
                    ^ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    uvloop/loop.c:169811:19: error: implicit declaration of function '_PyGen_Send' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
                ret = _PyGen_Send((PyGenObject*)yf, value == Py_None ? NULL : value);
                      ^
    uvloop/loop.c:169811:17: warning: incompatible integer to pointer conversion assigning to 'PyObject *' (aka 'struct _object *') from 'int' [-Wint-conversion]
                ret = _PyGen_Send((PyGenObject*)yf, value == Py_None ? NULL : value);
                    ^ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    uvloop/loop.c:169895:19: error: implicit declaration of function '_PyGen_Send' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
                ret = _PyGen_Send((PyGenObject*)yf, NULL);
                      ^
    uvloop/loop.c:169895:17: warning: incompatible integer to pointer conversion assigning to 'PyObject *' (aka 'struct _object *') from 'int' [-Wint-conversion]
                ret = _PyGen_Send((PyGenObject*)yf, NULL);
                    ^ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    3 warnings and 3 errors generated.
    error: command '/usr/bin/clang' failed with exit code 1
    ----------------------------------------
ERROR: Command errored out with exit status 1: /Users/robd/code/allfactors-sanic/env/bin/python3.10 -u -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'/private/var/folders/_1/hjn676yd5kq4ym6ksh47ns8h0000gn/T/pip-install-bt4385ls/uvloop_5ce278aa65a2427aa3ecbb25f01a5a84/setup.py'"'"'; __file__='"'"'/private/var/folders/_1/hjn676yd5kq4ym6ksh47ns8h0000gn/T/pip-install-bt4385ls/uvloop_5ce278aa65a2427aa3ecbb25f01a5a84/setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record /private/var/folders/_1/hjn676yd5kq4ym6ksh47ns8h0000gn/T/pip-record-hgw3w18r/install-record.txt --single-version-externally-managed --compile --install-headers /Users/robd/code/allfactors-sanic/env/include/site/python3.10/uvloop Check the logs for full command output.
```

I think this has to do with Python 3.10+ dropping `_PyGen_Send()` and requiring the use of `PyIter_Send()` now. Here's the issue tracking from Python: https://bugs.python.org/issue41756

You'll have no issues if you manually install via setup.py install

**Comment by fantix** [comment]
This is likely a similar issue as in #356 - re-cythonizing would fix it.

**Comment by 1st1** [comment]
Should be resolved with 0.16.0
